tutorials/interactive/index.md

```diff
@@ -0,0 +1,804 @@
+* The state object contains the widget's state, and the widget's `build()`
+  method.
+* The state object calls `setState()`,
+  telling the framework to redraw the widget.
+
+</div>
+
+In this section, you'll create a custom stateful widget.
+You'll replace two stateless widgets,
+the solid red star and the numeric count next to it, with a single custom
+stateful widget that manages a row with two children widgets: an IconButton
+and Text.
+
+A custom stateful widget requires creating two classes:
+
+1. A subclass of StatefulWidget that defines the widget.
+2. A subclass of State contains the state for that widget and defines
+   the widget's `build()` method.
+
+This section shows you how to build a StatefulWidget, called FavoriteWidget,
+for the Lakes app. The first step is choosing how the State is managed.
+
+<a name="step-1"></a>
+### Step 1: Decide who manages the widget's state
+
+There are different approaches to managing a widget's state, but in our example
+the widget itself, FavoriteWidget, will manage its own state.
+In this example, toggling the star is an isolated action that doesn't
+affect the parent widget or the rest of the UI,
+so the widget can handle its state internally.
+
+Learn more about the separation of widget and state,
+and how state might be managed, in [Managing state](#managing-state).
+
+<a name="step-2"></a>
+### Step 2: Subclass StatefulWidget
+
+The FavoriteWidget class manages its own state, so it overrides
+`createState` to create the State object.
+The framework calls `createState` when it wants to build the widget.
+In this example, `createState` creates an instance of _FavoriteWidgetState,
+which you'll implement in the next step.
+
+<!-- _code/lakes-interactive/main.dart -->
+<!-- skip -->
+{% prettify dart %}
+class FavoriteWidget extends StatefulWidget {
+  @override
+  _FavoriteWidgetState createState() => new _FavoriteWidgetState();
+}
+{% endprettify %}
+
+<aside class="alert alert-info" markdown="1">
+**Note:**
+Variables or classes that start with an underscore (_) are private.
+For more information, see [Libraries and
+visibility,](https://www.dartlang.org/guides/language/language-tour#libraries-and-visibility)
+a section the
+[Dart language tour.](https://www.dartlang.org/guides/language/language-tour)
+</aside>
+
+<a name="step-3"></a>
+### Step 3: Subclass the State object
+
+The custom State class stores the mutable information&mdash;the logic and
+internal state that can change over the lifetime of the widget.
+When the app first launches, the UI displays a solid red star,
+indicating that the lake has "favorite" status, and has a total of 41 “likes”.
+The state object stores this information in the
+`_isFavorited` and `_favoriteCount` variables.
+
+The state object also defines the `build` method. This `build` method
+creates a row containing a red IconButton, and Text.  The widget uses
+[IconButton,](https://docs.flutter.io/flutter/material/IconButton-class.html)
```

Comment:
nit: here and elsewhere, I would recommend putting the comma after the `[]()` sequence, so that there's no risk that the reader will think the comma is part of the class name.